modules/node-role-management.adoc

```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * machine_management/managing-infrastructure-machinesets.adoc
+
+[id="node-role-management_{context}"]
+= Node role management
+
+By default, all nodes, excluding those on the control plane, are labeled with the `node-role.kubernetes.io/worker: ""` role. Nodes with just the `worker` role are called worker, or compute nodes. To create a specialized non-compute node that is not on the control plane, like an infrastructure or app node, it is recommended to keep the `worker` role and define the additional role too. For example, for an infra node, it should be labeled with the `node-role.kubernetes.io/worker: ""` and `node-role.kubernetes.io/infra: ""` roles. You can remove the `worker` role and only apply the specialized node role, but that requires additional steps for upgrading due to constraints with the Machine Config Operator (MCO).
```

Comment:
Not sure what this last sentence means? It is allowed during upgrades to have custom pools no?
https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md#understanding-custom-pool-updates

Reply:
@kikisdeliveryservice The current issues that have been brought up in BZs are that customers want to strip the `worker` label and replace with an `infra` label. These doc updates are striving to recommend keeping the `worker` label, and also applying the `infra` label so they are both there. This scenario, as I've been told anyways, would not require a custom pool for upgrades.

What the last sentence is trying to say is that if you do remove the `worker` label, you'll need to take the additional steps to create the custom pool for upgrade purposes.

Can you confirm that is correct?

Reply:
Yes, basically, a node needs to always be a member of some Machine Config Pool to be managed by the MCO. The MCO doesn't touch any nodes that aren't members of a MCP.